servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/NettyHttpServer.java

```diff
@@ -346,6 +350,24 @@ private Completable handleRequestAndWriteResponse(final Single<StreamingHttpRequ
                             if (drainRequestPayloadBody) {
                                 payloadSubscribed.set(true);
                             }
+                            // Return 100 (Continue) when users subscribe to the request payload body.
+                            if (continueSent != null) {
+                                synchronized (continueSent) {
```

Comment:
Open for suggestions on how to avoid using `synchronized` block here to make sure we send 100 (Continue) only before the real response.